File: src/Unidesk/Unidesk/Dtos/ThesisOutcomeDto.cs
using System.ComponentModel.DataAnnotations;
using FluentValidation;
using Unidesk.Validations;

namespace Unidesk.Dtos;

public class ThesisOutcomeDto : TrackedEntityDto, IValidatedEntity<ThesisOutcomeDto>
{
    [Required]
    public string NameEng { get; set; }

    [Required]
    public string NameCze { get; set; }

    [Required]
    public string? DescriptionEng { get; set; }

    [Required]
    public string? DescriptionCze { get; set; }

    public void ValidateAndThrow(ThesisOutcomeDto item) => new ThesisOutcomeDtoValidation().ValidateAndThrow(item);
}

File: src/Unidesk/Unidesk.Client/src/api-client/models/ThesisOutcomeDto.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { TrackedEntityDto } from './TrackedEntityDto';

export type ThesisOutcomeDto = (TrackedEntityDto & {
nameEng: string;
nameCze: string;
descriptionEng?: string | null;
descriptionCze?: string | null;
});
